scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/license/LicenseSchedulingPolicy.java

```diff
@@ -186,17 +187,30 @@ private boolean isTaskExecutableUsingLicense(EligibleTaskDescriptor task) {
 
             // Do not execute if one of the required license can not be obtained
             final String[] requiredLicensesArray = requiredLicenses.split(",");
-            for (int i = 0; i < requiredLicensesArray.length; i++) {
+            final int nbRequiredLicenses = requiredLicensesArray.length;
+            boolean[] jobAlreadyUseLicense = new boolean[nbRequiredLicenses];
+
+            for (int i = 0; i < nbRequiredLicenses; i++) {
 
                 final String currentRequiredLicense = requiredLicensesArray[i];
+
+                // If the license is already obtained at the job level, use it without getting a new one
+                LinkedBlockingQueue<JobDescriptorImpl> eligibleJobsDescriptorsLicense = eligibleJobsDescriptorsLicenses.get(currentRequiredLicense);
+                if (containsJob(eligibleJobsDescriptorsLicense,
+                                ((EligibleTaskDescriptorImpl) task).getInternal().getJobId())) {
+                    jobAlreadyUseLicense[i] = true;
+                    continue;
+                }
+
                 if (!canGetTaskLicense(currentRequiredLicense)) {
                     logger.debug("License for " + currentRequiredLicense + " not available, keep task pending");
                     return false;
                 }
             }
             // Can be executed! Give it all required software licenses
-            for (int i = 0; i < requiredLicensesArray.length; i++) {
-                eligibleTasksDescriptorsLicenses.get(requiredLicensesArray[i]).add(task);
+            for (int i = 0; i < nbRequiredLicenses; i++) {
+                if (!jobAlreadyUseLicense[i])
```

Comment:
In the java convention a condition has always brackets, it improves the readability.

Reply:
Ok